mkt/extensions/models.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+import os.path
+
+from django.conf import settings
+from django.db import models
+from django.utils.functional import lazy
+
+import commonware.log
+from django_extensions.db.fields.json import JSONField
+
+from mkt.constants.base import (STATUS_CHOICES, STATUS_NULL, STATUS_PENDING,
+                                STATUS_PUBLIC)
+from mkt.extensions.utils import ExtensionParser
+from mkt.files.models import cleanup_file, nfd_str
+from mkt.translations.fields import save_signal, TranslatedField
+from mkt.translations.utils import to_language
+from mkt.site.models import ModelBase
+from mkt.site.storage_utils import copy_stored_file, private_storage
+from mkt.site.utils import smart_path
+from mkt.webapps.models import clean_slug
+
+
+log = commonware.log.getLogger('z.extensions')
+
+
+def _make_language_choices(languages):
+    return [(to_language(lang_code), lang_name)
+            for lang_code, lang_name in languages.items()]
+
+
+LANGUAGE_CHOICES = lazy(_make_language_choices, list)(settings.LANGUAGES)
+
+
+class Extension(ModelBase):
+    # Fields for which the manifest is the source of truth - can't be
+    # overridden by the API.
+    default_language = models.CharField(default=settings.LANGUAGE_CODE,
```

Comment:
i'm sure we need it, but i'm wondering what cases would we need attach languages to FxOS addons?